File: advent-of-code/bin/day02/day-02.pl
#!/usr/bin/perl -w

use strict;
use warnings;

sub input {
  my @list;
  open IN, '<', 'day-02.txt' or die $!;
  while ( <IN> ) {
    chomp;
    push @list, [ split( "x", $_ ) ];
  }
  close IN;
  return \@list;
}

sub paperRequired {
  my ( $x, $y, $z ) = sort { $a <=> $b } @_;
  return ( 2*$x*$y + 2*$x*$z + 2*$y*$z + $x*$y );
}

sub ribbonRequired {
  my ( $x, $y, $z ) = sort { $a <=> $b } @_;
  return ( 2*$x + 2*$y + $x*$y*$z );
}

my $sizes = input();
my $total_paper = 0;
my $total_ribbon = 0;
foreach ( @$sizes ) {
  $total_paper += paperRequired( @$_ );
  $total_ribbon += ribbonRequired( @$_ );
}

print $total_paper, " Square Feet of Wrapping Paper\n";
print $total_ribbon, " Linear Feet of Ribbon\n";

File: advent-of-code/src/day02/day-02.py
#!/usr/bin/env python

def paperRequired( x, y, z ):
  x,y,z = sorted( [ x, y, z ] )
  return ( 2*x*y + 2*x*z + 2*y*z + x*y )

def ribbonRequired( x, y, z ):
  x,y,z = sorted( [ x, y, z ] )
  return ( 2*x + 2*y + x*y*z )

data = []
with open( "day-02.txt", "r" ) as input_file:
  for line in input_file:
    data.append( line.rstrip().split("x") )

total_paper  = sum( map( lambda n: paperRequired( int(n[0]), int(n[1]), int(n[2]) ), data ) )
total_ribbon = sum( map( lambda n: ribbonRequired( int(n[0]), int(n[1]), int(n[2]) ), data ) )
  
print total_paper, "Square feet of wrapping paper";
print total_ribbon, "Linear feet of ribbon";

